e2e/testcases/reconciler_manager_test.go

```diff
@@ -281,18 +292,27 @@ func TestManagingReconciler(t *testing.T) {
 
 	// test case 4: the reconciler-manager should update the reconciler Deployment if the manifest in the ConfigMap has been changed.
 	nt.T.Log("Update the Deployment manifest in the ConfigMap")
-	nt.MustKubectl("apply", "-f", "../testdata/reconciler-manager-configmap-updated.yaml")
+	mustUpdateReconcilerTemplateConfigMap(nt, func(d *appsv1.Deployment) {
+		for i, container := range d.Spec.Template.Spec.Containers {
+			if container.Name == reconcilermanager.Reconciler {
+				d.Spec.Template.Spec.Containers[i].ImagePullPolicy = updatedImagePullPolicy
+			}
+		}
+	})
 	nt.T.Log("Restart the reconciler-manager to pick up the manifests change")
 	nomostest.DeletePodByLabel(nt, "app", reconcilermanager.ManagerName, true)
 	// Reset the reconciler-manager in the cleanup stage so other test cases can still run in a shared testing cluster.
 	nt.T.Cleanup(func() {
```

Comment:
nit: register cleanup before the mutation (`mustUpdateReconcilerTemplateConfigMap`) so that the cleanup is called in the case of an intermediate error

Reply:
done